client/src/RTCPeerConnector.ts

```diff
@@ -2,138 +2,138 @@ import User from "./User";
-						receiverId: user.id,
-						kind: "offer",
-					});
-					socket.emit("OFFER", JSON.stringify(offerPack));
-					onOfferSent(offerPack);
-				}
-			})
-			.catch((e) => {
-				console.error(e);
-			});
-	});
+  // for each user
+  users.forEach((user) => {
+    // emit an offer to the server to be broadcasted
+    user.peerConnection
+      .createOffer()
+      .then((offer) => {
+        return user.peerConnection.setLocalDescription(offer);
+      })
+      .then(() => {
+        if (user.peerConnection.localDescription) {
+          const offerPack = Pack({
+            sdp: user.peerConnection.localDescription,
+            senderId: socket.id,
+            receiverId: user.id,
+            kind: "offer",
+          });
+          socket.emit("OFFER", JSON.stringify(offerPack));
+          onOfferSent(offerPack);
+        }
+      })
+      .catch((e) => {
+        console.error(e);
+      });
+  });
 };
 
 export const openOfferListener = (
-	users: User[],
-	socket: Socket,
-	onOfferReceived: (Pack) => void = defaultOn,
-	onAnswerEmitted: (Pack) => void = defaultOn
+  users: User[],
+  socket: Socket,
+  onOfferReceived: (Pack) => void = defaultOn,
+  onAnswerEmitted: (Pack) => void = defaultOn
 ): void => {
-	// emit an answer when offer is received
-	socket.on("OFFER", (dataString) => {
-		const offerPack = JSON.parse(dataString);
-		const sender = findUserById(users, offerPack.senderId);
-		if (sender) {
-			onOfferReceived(offerPack);
-			// identify and use RTCPeerConnection object for the sender user
-			const peerConnection = (sender as User).peerConnection;
-			peerConnection
-				.setRemoteDescription(new RTCSessionDescription(offerPack.sdp)) // set remote description as the sender's
-				.then(() => {
-					peerConnection
-						.createAnswer()
-						.then((answer) => {
-							return peerConnection.setLocalDescription(answer);
-						})
-						.then(() => {
-							if (peerConnection.localDescription) {
-								const answerPack = Pack({
-									sdp: peerConnection.localDescription,
-									senderId: socket.id,
-									receiverId: offerPack.senderId,
-									kind: "answer",
-								});
-								socket.emit("ANSWER", JSON.stringify(answerPack));
-								onAnswerEmitted(answerPack);
-							}
-						})
-						.catch((e) => {
-							console.error(e);
-						});
-				})
-				.catch((e) => {
-					console.error(e);
-				});
-		} else {
-			console.error("Could not emit answer. Sender was not found on users list.");
-		}
-	});
+  // emit an answer when offer is received
+  socket.on("OFFER", (dataString) => {
+    const offerPack = JSON.parse(dataString);
+    const sender = findUserById(users, offerPack.senderId);
+    if (sender) {
+      onOfferReceived(offerPack);
+      // identify and use RTCPeerConnection object for the sender user
+      const peerConnection = (sender as User).peerConnection;
+      peerConnection
+        .setRemoteDescription(offerPack.sdp) // set remote description as the sender's
```

Comment:
This is the only change I actually made

Reply:
Didn't realize this was a thing.
Reference: https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/setRemoteDescription